src/strcat.rs

```diff
@@ -0,0 +1,29 @@
+//! Rust implementation of C library function `strcat`
+//!
+//! Licensed under the Blue Oak Model Licence 1.0.0
+
+use crate::CChar;
+
+/// Rust implementation of C library function `strcat`. Passing NULL
+/// (core::ptr::null()) gives undefined behaviour.
+#[cfg_attr(feature = "strcat", no_mangle)]
+pub unsafe extern "C" fn strcat(dest: *mut CChar, src: *const CChar) -> *const CChar {
+	crate::strcpy::strcpy(dest.add(crate::strlen::strlen(dest)), src);
+	dest
+}
+
+#[cfg(test)]
+mod test {
+	use super::*;
+
+	#[test]
+	fn simple() {
+		let mut dest = *b"hello\0";
```

Comment:
This is a buffer overflow. Try running this test in valgrind? You'll need to pad dest with a lot of NUL bytes to make space for the additions.